base/card-api.gts

```diff
@@ -334,36 +383,36 @@ export function serializeCard<CardT extends CardConstructor>(
     if (primitive in field.card) {
       attributes[fieldName] = serializedGet(model, fieldName);
     } else {
-      let nestedCard = serializedGet(model, fieldName);
-      if (field.containsMany && Array.isArray(nestedCard)) {
-        // TODO need to work thru how to represent a polymorphic contains many field's card refs.
-        // for now we are assuming that all cards in the containsMany field are the same card type
-        attributes[fieldName] = nestedCard.map(resource => resource.attributes);
-        if (nestedCard.length === 0) {
-          continue;
+      if (field.fieldType === 'containsMany') {
+        let collection = model[fieldName as keyof Card];
+        if (!Array.isArray(collection)) {
+          throw new Error(`expected containsMany field '${fieldName}' value in card '${model.constructor.name}' to be an array`);
+        }
+        let fieldValue: any[] = [];
+        let metaFieldItems: MetaFieldItem[] = [];
+        let fieldIdentity = Loader.identify(field.card);
+        if (!fieldIdentity) {
+          throw new Error(`bug: encountered a card that has no Loader identity: ${field.card.name}`);
+        }
+        for (let index of collection.keys()) {
+          let { value, metaFieldItem } = resourcePartsFor(model, fieldName, index);
+          fieldValue.push(value);
+          if (metaFieldItem) {
+            metaFieldItems.push(metaFieldItem);
+          } else {
+            metaFieldItems.push({ adoptsFrom: fieldIdentity });
+          }
+        }
+        attributes[fieldName] = fieldValue;
+        if (metaFieldItems.length > 0 && metaFieldItems.some(i => !isEqual(i, { adoptsFrom: fieldIdentity }))) {
+          fields = { ...fields, ...{ [fieldName]: metaFieldItems } };
+        }
+      } else if (field.fieldType === 'contains') {
+        let { value, metaFieldItem } = resourcePartsFor(model, fieldName);
+        attributes[fieldName] = value;
+        if (metaFieldItem) {
+          fields = { ...fields, ...{ [fieldName]: metaFieldItem } };
```

Comment:
refactor this such that our `Field.serialize()` interface is capable of returning a separate meta object as part of its return value. this should internalize the arrayness for `containsMany` fields so that it doesn't leak out here